### PR TITLE
Other minor improvements for toric varieties

### PR DESCRIPTION
**Pull request by HereAround**
- Mention in documentation that the Cox ring is also known as total coordinate ring.
- Add a constructor for the `toric_affine_space`.
- Cache the toric ideal of a `NormalToricVariety` which is affine.
- Replace `abelian_group` by `free_abelian_group` whenever possible.

This PR contains the changes in https://github.com/oscar-system/Oscar.jl/pull/929 for completeness.

### CODE REVIEW

**Comment by fingolfin** [comment]
PR #929 was merged, so please rebase.

File: src/ToricVarieties/NormalToricVarieties/attributes.jl [hunk 1]
```diff
@@ -56,7 +56,8 @@ export euler_characteristic
 @doc Markdown.doc"""
     cox_ring(v::AbstractNormalToricVariety)
 
-Return the Cox ring of the normal toric variety `v`.
+Computes the Cox ring of the normal toric variety `v`.
+Note that [CLS11](@cite) refers to this ring as the `total coordinate ring`.
```
Comment (fingolfin) [inline src/ToricVarieties/NormalToricVarieties/attributes.jl]: This is a quote, not code, so:
```suggestion
Note that [CLS11](@cite) refers to this ring as the "total coordinate ring".
```

File: src/ToricVarieties/NormalToricVarieties/constructors.jl [hunk 2]
```diff
@@ -187,6 +187,47 @@ end
...
+        m[i,i] = 1
+    end
+    C = positive_hull(m)
+    
+    # construct the variety
+    fan = PolyhedralFan(C)
+    pmntv = Polymake.fulton.NormalToricVariety(Oscar.pm_object(fan))
+    variety = NormalToricVariety(pmntv, Dict())
+    
+    # set known properties´
```
Comment (fingolfin) [inline src/ToricVarieties/NormalToricVarieties/constructors.jl]:
```suggestion
    # set known properties
```

File: src/ToricVarieties/NormalToricVarieties/constructors.jl [hunk 3]
```diff
@@ -221,13 +262,13 @@ function toric_projective_space(d::Int)
...
-    set_attribute!(variety, :character_lattice, abelian_group([0 for i in 1:d]))
-    set_attribute!(variety, :torusinvariant_divisor_group, abelian_group([0 for i in 1:d+1]))
+    set_attribute!(variety, :character_lattice, free_abelian_group(d))
+    set_attribute!(variety, :torusinvariant_divisor_group, free_abelian_group(d+1))
     set_attribute!(variety, :map_from_cartier_divisor_group_to_torus_invariant_divisor_group, Hecke.identity_map(torusinvariant_divisor_group(variety)))
     set_attribute!(variety, :map_from_cartier_divisor_group_to_picard_group, map_from_weil_divisors_to_class_group(variety))
     set_attribute!(variety, :stanley_reisner_ideal, ideal([prod(Hecke.gens(cox_ring(variety)))]))
     set_attribute!(variety, :irrelevant_ideal, ideal(Hecke.gens(cox_ring(variety))))
-    betti_numbers = [if iseven(i) fmpz(1) else fmpz(0) end for i in 0:2*d]
+    betti_numbers = [fmpz(1) for i in 0:d]
```
Comment (fingolfin) [inline src/ToricVarieties/NormalToricVarieties/constructors.jl]: Could also do this (tiny bit faster / uses less memory -- but in reality it doesn't matter here, so pick what your taste prefers)
```suggestion
    betti_numbers = fill(fmpz(1), d+1)
```

**Review by fingolfin — APPROVED** [review]

File: src/ToricVarieties/NormalToricVarieties/constructors.jl (hunk 3 above)
Comment (HereAround) [reply src/ToricVarieties/NormalToricVarieties/constructors.jl]: Thank you. Done in https://github.com/oscar-system/Oscar.jl/pull/940/commits/83fa4c7b9528f694fc8da99b125cbb15abe57319. (After the rebase, I could not commit your suggestion directly.)

**Comment by HereAround** [comment]
> PR #929 was merged, so please rebase.

Done.

**Comment by HereAround** [comment]
All tests green - that is how I like it. Please consider merging if you are satisfied.

**Review by fingolfin — APPROVED** [review]

**Pull request merged**